core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -144,69 +145,56 @@ private fun computePaths(
-                            )
-                        )
-                    }
-                    LoadingGaugeConstraints::class.java -> {
-                        throw NoPathFoundException(
-                            IncompatibleConstraintsPathResponse(
-                                relaxedPathResponse,
-                                IncompatibleConstraints(
-                                    listOf(),
-                                    listOf(Pathfinding.Range(Offset.zero(), Offset.zero())),
-                                    listOf()
-                                )
-                            )
-                        )
+    if (
+        possiblePathWithoutErrorNoConstraints != null &&
+            possiblePathWithoutErrorNoConstraints.ranges.isNotEmpty()
+    ) {
+        val elecRanges = mutableListOf<RangeValue<String>>()
+        val gaugeRanges = mutableListOf<Pathfinding.Range<TravelledPath>>()
+        val signalRanges = mutableListOf<RangeValue<String>>()
+        var travelledPathBlockStart =
+            Offset<TravelledPath>(
+                -possiblePathWithoutErrorNoConstraints.ranges.first().start.distance
+            )
+        for (range in possiblePathWithoutErrorNoConstraints.ranges) {
+            for (currentConstraint in constraints) {
+                for (blockedRange in currentConstraint.apply(range.edge)) {
+                    if (blockedRange.end < range.start || range.end < blockedRange.start) {
+                        // The blocked range is outside the used part
+                        continue
                     }
-                    SignalingSystemConstraints::class.java -> {
-                        throw NoPathFoundException(
-                            IncompatibleConstraintsPathResponse(
-                                relaxedPathResponse,
-                                IncompatibleConstraints(
-                                    listOf(),
-                                    listOf(),
-                                    listOf(
-                                        RangeValue(
-                                            Pathfinding.Range(Offset.zero(), Offset.zero()),
-                                            "signal"
-                                        )
-                                    )
-                                )
-                            )
+                    val range =
+                        Pathfinding.Range(
+                            travelledPathBlockStart +
+                                Offset.max(blockedRange.start, range.start).distance,
+                            travelledPathBlockStart +
+                                Offset.min(blockedRange.end, range.end).distance
                         )
+                    when (currentConstraint::class.java) {
+                        ElectrificationConstraints::class.java -> {
+                            elecRanges.add(RangeValue(range, "elec"))
```

Comment:
I don't remember if it was discussed in the ticket, but this means we potentially return ranges of constraints that could be merged, i.e.: [100, 200, "15000V], [200, 300, "15000V"] => [100, 300, "15000V"]. I feel like it would be useful to merge them when possible, but I'll let you check if it's required and a change we'd want.